Initial 10.13 High Sierra Deploy

mircoianese opened the pull request:
Hello.
I have edited Deploy.sh to fix some High Sierra issue I have encountered and to correctly patch some other things. Please have a look and report back if you have any issue.

Thank you! 

**Note:** Actual PixelClock Patch for QHD Display breaks Clipboard on High Sierra. This script uses CoreDisplayFixUp.kext + Lilu.kext (both in Clover/kexts). If you already have patched with previous Deploy, you need to restore the Vanilla CoreDisplay.framework to make the Clipboard work again.
If you clean install and run directly this new Deploy then you shouldn't experience this issue.

**Note 2:** Final Deploy is in last commit. Deploy in the previous one had one bug.

Commit:
Initial 10.13 High Sierra Update

Commit:
Initial 10.13 High Sierra Update

Commit:
Initial 10.13 High Sierra Update

Commit:
Initial 10.13 High Sierra Update

Commit:
Initial 10.13 High Sierra Update

mircoianese committed:
Fixed script issue

mircoianese committed:
Fixed Script Issue

mircoianese committed:
Fixed Script Issue

mircoianese committed:
Fixed if clause

mircoianese committed:
Delete USBX and PNLF as they were already patched

mircoianese committed:
Delete USBX and PNLF as they were already patched

mircoianese committed:
Delete USBX and PNLF as they were already patched

mircoianese committed:
Update README.md

mircoianese committed:
Fix Deploy

mircoianese committed:
Fix Deploy

mircoianese committed:
Changed git link in ReadMe

git link was referring to Syscl git, it may cause confusion

mircoianese committed:
Update README.md

ZombieTheBest commented:
@chenyisen tested the script and reported all the issue he got. Script is now fixed and should work correctly on both Sierra and High Sierra.

Please report back any other issue you may encounter.

vinv commented:
Hi,

I currently have Sierra and have run syscl's Deploy script. If I wanted to try yours, do I first install High Sierra (via the App Store), then run your script or is there a different recommended upgrade procedure?

Vin

ZombieTheBest commented:
Upgrade First then run my modified script. Of course always backup your
data first. Also remember that installer will convert your partition in
APFS, upgrade Clover and put APFS.efi in drivers64 (you can find it in my
clover folder). Also, use my clover folder in pendrive if you can't boot
anymore after upgrade!

Il 26 Ott 2017 19:12, "vinv" <notifications@github.com> ha scritto:

> Hi,
>
> I currently have Sierra and have run syscl's Deploy script. If I wanted to
> try yours, do I first install High Sierra (via the App Store), then run
> your script or is there a different recommended upgrade procedure?
>
> Vin
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/syscl/XPS9350-macOS/pull/156#issuecomment-339735331>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVqfzpKR9ng_xjw9bjgucgQzhIZkX7AGks5swL2XgaJpZM4QEbyP>
> .
>

ZombieTheBest commented:
@vinv Also please if you could provide Deploy.sh log and you Clover folder after patching the system that would be really appreciated. Just to check if the script worked correctly!

Thank you!

syscl commented:
Thank you for your effort! But I haven’t experienced Pixel Clock patch break clipboard issue on 10.13 and not even happened on my M3800 either. Do you have any links indicating this patch breaks clipboard on 10.13? 

Thank you in advanced,
syscl

ZombieTheBest commented:
Script is simple using CoreDisplayFixUp instead of Pixel Clock Patch. I am
100% sure that is the Pixel Patch. You can see in the post I made in the
issues that even another user was experiencing that.
https://github.com/syscl/XPS9350-macOS/issues/157

As you can see, in the last version of Deploy (when I fixed the if clause),
after clean install, he was not experiencing it anymore, so it must be the
Pixel Clock Patch. As you can see my edited script just checks for the OS
version and then, if it's 10.13, it does patch the other way.

I will leave here link to my discussion on tonymac too:
https://www.tonymacx86.com/threads/10-13-ssdt-usb-patching-alc256-audio-reboot.235231/page-4

Thank you

Il 27 Ott 2017 09:50, "Zhou Yating" <notifications@github.com> ha scritto:

> Thank you for your effort! But I haven’t experienced Pixel Clock patch
> break clipboard issue on 10.13 and not even happened on my M3800 either. Do
> you have any links indicating this patch breaks clipboard on 10.13?
>
> Thank you in advanced,
> syscl
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/syscl/XPS9350-macOS/pull/156#issuecomment-339899582>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVqfzm62tcI154aH18KO25T8-IpPbAq2ks5swYtUgaJpZM4QEbyP>
> .
>

syscl commented:
Thank you, since I have Iris model this I never worry about this patch but I have found a new one could you please help me to see if it works or not? 

Thank you in advanced,
syscl

ZombieTheBest commented:
I can test it for you this evening. Btw right now CoreDisplayFixUp+Lilu is
working good for me

Il 27 Ott 2017 10:10, "Zhou Yating" <notifications@github.com> ha scritto:

Thank you, since I have Iris model this I never worry about this patch but
I have found a new one could you please help me to see if it works or not?

Thank you in advanced,
syscl

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/syscl/XPS9350-macOS/pull/156#issuecomment-339903676>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AVqfzrcE_CmsbwT02_BUhNS7M8Dn6cdaks5swY_vgaJpZM4QEbyP>
.

syscl commented:
Remove CoreDisplayFixup and try this patch to see if Clipboard corrupt or not

```sh
sudo perl -i.bak -pe 's|\xB8\x01\x00\x00\x00\xF6\xC1\x01\x0F\x85|\x48\x33\xC0\x90\x90\x90\x90\x90\x90\xE9|sg' /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay
sudo codesign -f -s - /System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay
```

Thank you in advanced,
syscl

ZombieTheBest commented:
@syscl gonna test it in 10-12 hours. Thank you :)
PS: in High Sierra I was experiencing Bluetooth issue. Solved by deleting
one kext in S/L/E. You can find that solution on my tonymac post too. This
was also added to the script, check it too!
I also updated the config.plist etc.

Il 27 Ott 2017 10:18, "Zhou Yating" <notifications@github.com> ha scritto:

Remove CoreDisplayFixup and try this patch to see if Clipboard corrupt or
not

sudo perl -i.bak -pe
's|\xB8\x01\x00\x00\x00\xF6\xC1\x01\x0F\x85|\x48\x33\xC0\x90\x90\x90\x90\x90\x90\xE9|sg'
/System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay
sudo codesign -f -s -
/System/Library/Frameworks/CoreDisplay.framework/Versions/Current/CoreDisplay

Thank you in advanced,
syscl

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/syscl/XPS9350-macOS/pull/156#issuecomment-339905506>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AVqfzvh6DjH-8h_ixMtzsA1m9CYWuBz4ks5swZHZgaJpZM4QEbyP>
.

ZombieTheBest commented:
@syscl That broke clipboard. Reverting back to CoreDisplayFixup

PMheart commented:
That's really weird then. I think CoreDisplayFixup is doing totally the same thing: https://github.com/PMheart/CoreDisplayFixup/blob/master/CoreDisplayFixup/IntelPatcher.cpp#L35

syscl commented:
pm no need to remove just move back is enough.

ZombieTheBest commented:
Could you send here your vanilla file? Cause restoring back my file is not
fixing. That's REALLY WEIRD. Before applying the patch everything was
working.

Il 27 Ott 2017 23:13, "Zhou Yating" <notifications@github.com> ha scritto:

> pm no need to remove just move back is enough.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/syscl/XPS9350-macOS/pull/156#issuecomment-340087158>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVqfzmtrnizGkK7Qmo0rcoyluWy2vKePks5swkd2gaJpZM4QEbyP>
> .
>

syscl commented:
@ZombieTheBest Here's the original file

[CoreDisplay.framework.zip](https://github.com/syscl/XPS9350-macOS/files/1423671/CoreDisplay.framework.zip)

syscl

PMheart commented:
If restoring to the vanilla one doesn't help either, try also rebuilding shared cache by
`sudo update_dyld_shared_cache -force`

ZombieTheBest commented:
@PMheart That fixed, thank you both! Btw I will continue to use CoreDisplayFixUp for now 👍

mircoianese committed:
Delete SSDT-EC.aml

mircoianese committed:
Added SSDT-PNLF for BackLight Injection

mircoianese committed:
Added SSDT-PNLF for BackLight Injection

mircoianese committed:
Update README.md

mircoianese committed:
Added SSDT-PNLF for BackLight Injection

mircoianese committed:
Update README.md

mircoianese committed:
Stable Deploy

mircoianese committed:
Stable Update

mircoianese committed:
Stable High Sierra Support

mircoianese committed:
Update README.md

mircoianese committed:
Update README.md

mircoianese committed:
Update README.md

mircoianese committed:
Fixed SSDTs Compile Error credits @squash- @zombiethebest

mircoianese committed:
Update README.md

syscl commented:
Hi Zombie, please do whatever you like this period I will look into it in the future. I am sorry that I still have a lot of projects and exams, and I am a TA/RA for operating system course as well. Life is busy for me these days. Sorry for my late reply.

syscl

mircoianese committed:
BETA Release to fix DSDT issue

maz-1 committed:
Fix a few mistakes in Deploy.sh

1. Fix a syntax error in the if statement of CoreDisplay patch function
2. Fix a error in the sysusbfix script, which doesn't store mounted usb drives correctly

maz-1 committed:
Install CoreDisplayFixup.ketx on iris models

mircoianese committed:
Merge pull request #12 from maz-1/beta

Fix a few mistakes in Deploy.sh

maz-1 committed:
Add SSDT-ALS0 to save brightness

maz-1 committed:
Sort SSDTs

mircoianese committed:
Merge pull request #13 from maz-1/beta

Add SSDT-ALS0.aml to save brightness (c) @maz-1

maz-1 committed:
Set proper HiDPIEFILogin and UIScale parameter

maz-1 committed:
Update config.plist

mircoianese committed:
Merge pull request #14(c) maz-1

Set proper EFILoginHiDPI and UIScale parameter (c) maz-1

mircoianese committed:
Update README.md

mircoianese committed:
Updated Kexts

mircoianese committed:
Fixed a lot of compile errors (c) ZombieTheBest

Ready to merge stable branch.

mircoianese committed:
Added PARSEOPIF Patch DSDT

mircoianese committed:
Resolved merge conflict

mircoianese committed:
Merge branch 'beta'

Merge BETA to Master

mircoianese committed:
Update README.md

mircoianese committed:
Updated Clover to Rehabman's Branch + Added AutoMerge

mircoianese committed:
Merge branch 'master' of https://github.com/ZombieTheBest/XPS9350-10.13-HighSierra

mircoianese committed:
Update README.md

mircoianese committed:
Update README.md

syscl commented:
Hi @maz-1 @ZombieTheBest , Iris model still require CoreDisplayPatch?

Thank you in advance,
syscl

maz-1 commented:
CoreDisplayPatch can help reducing boot logo glitch to minimum

syscl commented:
@maz-1 no, that’s not CoreDisplay patch reduces the boot glitches but CSM and framebuffer that fix the boot glitches. 

Iris Model did not need CoreDisplay patch.

maz-1 commented:
Fine, pick commits you need then

AurelienKun commented:
Is this abandonned ?

syscl commented:
@AurelienKun  This is not abandon, but seems I cannot merge this PR @mircoianese

mircoianese commented:
> @AurelienKun This is not abandon, but seems I cannot merge this PR @mircoianese

Hi, this is a super-late response but had no notifications on this thread.
Unfortunately I do not own this laptop anymore so I can't help with this. I have also forgot most of the things that were done in this PR. 

Sorry I can't help

liuzhu007 commented:
您好：
　　我已收到您的来信，容我拜读后与您联系！
　谢谢！　　　　　　　　　　　　　　　　　　                                 刘铸
